pkgs/servers/x11/xorg/overrides.nix

```diff
@@ -759,9 +759,6 @@ self: super:
     ];
     propagatedBuildInputs = [ self.xauth ]
                          ++ lib.optionals isDarwin [ self.libX11 self.xorgproto ];
-    prePatch = ''
-      sed -i 's|^defaultserverargs="|&-logfile \"$HOME/.xorg.log\"|p' startx.cpp
-    '';
```

Comment:
I think this should be part of the commit were you patched xorg.

Reply:
It's one logical step: Old solution with hard-coded path in xinit -> new solution with upstream path and configurable override in NixOS that was set up in the previous commits.